torchaudio/datasets/yesno.py

```diff
@@ -1,150 +1,45 @@
-            and returns a transformed version. E.g, ``transforms.Spectrogram``. (
-            Default: ``None``)
-        target_transform (Callable, optional): A function/transform that takes in the
-            target and transforms it. (Default: ``None``)
-        download (bool, optional): If true, downloads the dataset from the internet and
-            puts it in root directory. If dataset is already downloaded, it is not
-            downloaded again. (Default: ``False``)
-        dev_mode(bool, optional): If true, clean up is not performed on downloaded
-            files.  Useful to keep raw audio and transcriptions. (Default: ``False``)
-    """
-    raw_folder = 'yesno/raw'
-    processed_folder = 'yesno/processed'
-    url = 'http://www.openslr.org/resources/1/waves_yesno.tar.gz'
-    dset_path = 'waves_yesno'
-    processed_file = 'yesno.pt'
-
-    def __init__(self, root, transform=None, target_transform=None, download=False, dev_mode=False):
-        self.root = os.path.expanduser(root)
-        self.transform = transform
-        self.target_transform = target_transform
-        self.dev_mode = dev_mode
-        self.data = []
-        self.labels = []
-        self.num_samples = 0
-        self.max_len = 0
 
-        if download:
-            self.download()
-
-        if not self._check_exists():
-            raise RuntimeError('Dataset not found.' +
-                               ' You can use download=True to download it')
-        self.data, self.labels = torch.load(os.path.join(
-            self.root, self.processed_folder, self.processed_file))
-
-    def __getitem__(self, index):
-        """
-        Args:
-            index (int): Index
+import torch.utils.data as data
 
-        Returns:
-            Tuple[torch.Tensor, int]: The output tuple (image, target) where target
-            is index of the target class.
-        """
-        audio, target = self.data[index], self.labels[index]
+import torchaudio
+from torchaudio.datasets.utils import download, extract, shuffle, walk
 
-        if self.transform is not None:
-            audio = self.transform(audio)
 
-        if self.target_transform is not None:
-            target = self.target_transform(target)
+def load_yesno_item(fileid, path, ext_audio):
+    # Read label
+    label = fileid.split("_")
 
-        return audio, target
+    # Read wav
+    file_audio = os.path.join(path, fileid + ext_audio)
+    waveform, sample_rate = torchaudio.load(file_audio)
 
-    def __len__(self):
-        return len(self.data)
+    return {"label": label, "waveform": waveform, "sample_rate": sample_rate}
 
-    def _check_exists(self):
-        return os.path.exists(os.path.join(self.root, self.processed_folder, self.processed_file))
 
-    def download(self):
-        """Download the yesno data if it doesn't exist in processed_folder already."""
-        from six.moves import urllib
-        import tarfile
+class YESNO(data.IterableDataset):
```

Comment:
Can we change this back into a regular data.Dataset to maintain backwards compatibility? Same for the other 3 datasets, I think they can all be indexable, which then works with the Samplers etc. we have in pytorch.

Reply:
Agreed. The datasets are naturally indexable, even without loading the data in memory.